### PR TITLE
Update replacement function

### PR DESCRIPTION
**Pull request by ataylorme**
## Description
Addresses issue #192 by adding a `PHPNamespace` option to `config/themeConfig.js` in order to allow PHP namespaces to be replaced.

PHP namespaces will be replaced first since they are the first key in `nameFieldDefaults` and `getStringReplacementTasks` runs `Object.keys` on `nameFieldDefaultsnameFieldDefaults` to generate the replacement order.

I remembered an [old comment from Tonya](https://github.com/wprig/wprig/pull/47#discussion_r201158200) about the `searchValue` setting of `string` with [`gulp-string-replace`](https://www.npmjs.com/package/gulp-string-replace) not replacing everything.

I dug into the code for the plugin and `string` [runs a normal `replace` instance](https://github.com/tomaszczechowski/gulp-string-replace/blob/1.1.2/index.js#L83), which replaces the first occurrence, not all occurrences, of the string.

This is problematic as the entire file contents are a single string and we need to replace all occurrences, not just the first.

The `regex` search method, on the other hand, [always does a global replacement](https://github.com/tomaszczechowski/gulp-string-replace/blob/1.1.2/index.js#L73).

Since the namespace string has a backslash it took some messing around with escaping as both JavaScript and regex patterns need backslashes to be escaped.

Long story short it is all sorted now and users can update the `PHPNamespace` in the config to have the namespace updated in the production build.

## List of changes
* Replace PHP Namespace with the `PHPNamespace` option in `config/themeConfig.js`
* Escape backslashes
* Move the string replace options to the getStringReplacementTasks util function
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] My code is tested.
- [x] I want my code added to WP Rig.

### CODE REVIEW

**Comment by ataylorme** [comment]
> I approve this as it is now, but we should be aware of it. Not sure we need to add the above now, but we need to keep it in mind (in JS it would be WP_Rig\\\\WP_Rig 🤣)

Backslashes have to be escaped again for regex. It could get pretty ridiculous. I'm going to merge this for now and we can consider it if double backslashes are used in WP Rig in the future.
